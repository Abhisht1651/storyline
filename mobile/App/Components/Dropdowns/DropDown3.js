// @flow

import React, { useEffect } from 'react'
import { View, Text } from 'react-native'
import { TouchableOpacity } from 'react-native-gesture-handler'
import styles from '../Styles/DropDownStyles/DropDown3Style'
import Elipsis from '../../Images/Icons/icon-elipsis.svg'
import useHooks from './DropDown3Hooks'

type Props = {
  headerTitle: string,
  content: Array<string>,
  getOption: (arg: number) => void,
  zIndex: number
}

const DropDown3 = (props: Props) => {
  const { hidden, clickHandler, toggleHandler, selected } = useHooks()
  const { headerTitle, content, getOption, zIndex } = props

  const shadowStyle = {
    shadowOpacity: 0.5,
    shadowRadius: 5,
    shadowColor: '#1F1F6016',
    shadowOffSet: { width: 0, height: 0 }
  }

  useEffect(() => {
    getOption(selected)
  }, [selected])

  const list = []
  for (let i = 0; i < content.length; i += 1) {
    if (i !== content.length - 1) {
      list.push(
        <TouchableOpacity
          onPress={() => clickHandler(i)}
          style={styles.dropDownWidth}
          activeOpacity={1}
          key={i}>
          <View style={styles.listItem}>
            <Text style={styles.itemText}>{content[i]}</Text>
          </View>
        </TouchableOpacity>
      )
    } else {
      list.push(
        <TouchableOpacity
          onPress={() => clickHandler(i)}
          style={styles.dropDownWidth}
          activeOpacity={1}
          key={i}>
          <View style={styles.listLastItem}>
            <Text style={styles.lastItemText}>{content[i]}</Text>
          </View>
        </TouchableOpacity>
      )
    }
  }

  return (
    <View style={styles.container} zIndex={zIndex}>
      <TouchableOpacity onPress={toggleHandler} activeOpacity={1} style={styles.iconOnly}>
        <Elipsis />
      </TouchableOpacity>

      {!hidden && (
        <View style={[styles.dropDownBody, shadowStyle]} elevation={4}>
          <View style={styles.dropDownHeader}>
            <View style={styles.headerItem}>
              <Text style={styles.headerText}>{headerTitle}</Text>
            </View>
            <TouchableOpacity onPress={toggleHandler} style={styles.dropDownIcon} activeOpacity={1}>
              <Elipsis />
            </TouchableOpacity>
          </View>
          {list}
        </View>
      )}
    </View>
  )
}

DropDown3.defaultProps = {
  headerTitle: 'Posts',
  content: ['Privacy', 'Edit post', 'Remove post'],
  zIndex: 1
}

export default DropDown3
