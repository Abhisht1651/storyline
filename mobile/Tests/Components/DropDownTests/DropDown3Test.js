import { shallow } from 'enzyme'
import React from 'react'
import { renderHook, act } from '@testing-library/react-hooks'

import useHooks from '../../../App/Components/Dropdowns/DropDown3Hooks'
import DropDown3 from '../../../App/Components/Dropdowns/DropDown3'

describe('DropDown3', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(<DropDown3 />)
      expect(component).toMatchSnapshot()
    })
  })
})

describe('DropDown3', () => {
  it('should hide/show the dropdown', () => {
    const { result } = renderHook(useHooks)
    const val = result.current.hidden
    act(() => {
      result.current.toggleHandler()
    })
    expect(result.current.hidden).toBe(!val)
  })
})
