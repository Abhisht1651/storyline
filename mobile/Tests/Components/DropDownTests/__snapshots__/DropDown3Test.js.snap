// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown3 Rendering should match to snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
    }
  }
  zIndex={1}
>
  <TouchableOpacity
    activeOpacity={1}
    delayLongPress={600}
    extraButtonProps={
      Object {
        "rippleColor": "transparent",
      }
    }
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "height": 16.428571428571427,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <SvgMock />
  </TouchableOpacity>
</View>
`;
